src/filters/truncate.js

```diff
@@ -2,30 +2,33 @@
 
 angular.module('openshiftCommonUI')
   .filter('truncate', function() {
-    return function(str, charLimit, useWordBoundary, newlineLimit) {
-      if (!str) {
+    return function truncate(str, charLimit, useWordBoundary, newlineLimit, removeExtraChars) {
+      if(typeof str !== 'string' || typeof charLimit !== 'number') {
         return str;
       }
 
       var truncated = str;
+      var limiter = charLimit;
+      var removedCharsLength = charLimit - (removeExtraChars || 0);
 
-      if (charLimit) {
-        truncated = truncated.substring(0, charLimit);
-      }
-
-      if (newlineLimit) {
+      if (newlineLimit && truncated.length > limiter) {
         var nthNewline = str.split("\n", newlineLimit).join("\n").length;
         truncated = truncated.substring(0, nthNewline);
       }
 
-      if (useWordBoundary !== false) {
-        // Find the last word break, but don't look more than 10 characters back.
-        // Make sure we show at least the first 5 characters.
-        var startIndex = Math.max(4, charLimit - 10);
-        var lastSpace = truncated.lastIndexOf(/\s/, startIndex);
-        if (lastSpace !== -1) {
-          truncated = truncated.substring(0, lastSpace);
+      if (truncated.length > limiter) {
+        if (useWordBoundary) {
+          var tmpLimiter = truncated.lastIndexOf(' ', limiter);
+
+          if (tmpLimiter > 0) {
+            limiter = tmpLimiter;
+          }
         }
+        truncated = truncated.substring(0, limiter);
+      }
+
+      if (removeExtraChars && truncated.length >= removedCharsLength) {
+        truncated = truncate(truncated, removedCharsLength, useWordBoundary);
```

Comment:
If I'm reading this right, won't this remove the entire string if I pass in a long unbroken identifier? Or am I misunderstanding?

Before it would never look more than 10 characters back for a word boundary.